kpi/filters.py

```diff
@@ -77,8 +77,15 @@ def filter_queryset(self, request, queryset, view):
         # the query to be processed right now. Otherwise, because queryset is
         # a lazy query, Django creates (left) joins on tables when queryset is
         # interpreted and it is way slower than running this extra query.
-        asset_ids = list(owned_and_explicit_shared.union(subscribed)
-                         .values_list('id', flat=True))
+
+        # TODO: Figure out a better way to get subscribed collection's children
+        # As a temporary fix we union a separate query for id's of subscribed
+        # collection parents, bypassing the broken `?q=parent__uid` query
+        asset_ids = list(
```

Comment:
we discussed that this fix is okay long-term, and that `?q=parent__uid` does work (because `SearchFilter` handles it), but a comment here about why it's necessary to check if the user has a subscription to the parent (and not just to the asset itself) would be helpful

Reply:
Changed the comment to:
```
# Since user would be subscribed to a collcetion and not
# the assets themselves, we append children of subscribed
# collections to the queryset in order for `?q=parent__uid`
# queries to return the collection's children
.union(queryset.filter(parent__in=subscribed).values("pk")
```

Reply:
@duvld, typo :-) `collcetion`